category_encoders/__init__.py

```diff
@@ -26,6 +26,21 @@
 from category_encoders.rankhot import RankHotEncoder
 from category_encoders.glmm import GLMMEncoder
 from category_encoders.quantile_encoder import QuantileEncoder, SummaryEncoder
+import sklearn
+import warnings
+from textwrap import dedent
+
+if sklearn.__version__ < '1.2.0':
```

Comment:
I don't really like the input warning here. I know of some users who use this library in their project and they try to suppress the warning for their end-users.  
Also, at the moment it also does not work anyway. 
I'd be an favor of not issuing a warning here but have something on the `index.rst` (c.f. below comment)

Reply:
I fully agree, in fact warnings are annoying. Let's remove this. Do you want me to add a new commit or you prefer to do it in the merge process?